File: common/src/main/java/io/github/notenoughupdates/moulconfig/gui/component/WhenComponent.java
package io.github.notenoughupdates.moulconfig.gui.component;

import io.github.notenoughupdates.moulconfig.gui.GuiComponent;

import java.util.function.Supplier;

public class WhenComponent extends IndirectComponent {
    public WhenComponent(Supplier<? extends Boolean> condition, Supplier<? extends GuiComponent> ifTrue, Supplier<? extends GuiComponent> ifFalse) {
        super(() -> condition.get() ? ifTrue.get() : ifFalse.get());
    }
}

File: common/src/main/java/io/github/notenoughupdates/moulconfig/gui/component/WhenComponent.kt
package io.github.notenoughupdates.moulconfig.gui.component

import io.github.notenoughupdates.moulconfig.gui.GuiComponent
import java.util.function.Supplier

class WhenComponent(
    val condition: Supplier<out Boolean>,
    val ifTrue: Supplier<out GuiComponent>,
    val ifFalse: Supplier<out GuiComponent>,
) : IndirectComponent(
    Supplier {
        if (condition.get())
            ifTrue.get()
        else
            ifFalse.get()
    }
)